org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/junit/launcher/MockLaunchConfig.java

```diff
@@ -0,0 +1,263 @@
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.core.ILaunchDelegate;
+
+import org.eclipse.jdt.internal.junit.launcher.JUnitLaunchConfigurationConstants;
+
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+
+class MockLaunchConfig implements ILaunchConfiguration {
+
+	private String fProjectName;
+	private String fTestRunnerKind;
+	private String fMainTypeName;
+	private String fContainerHandle;
+
+	public void setProjectName(String projectName) {
+		fProjectName= projectName;
+	}
+
+	public void setTestRunnerKind(String testRunnerKind) {
+		fTestRunnerKind= testRunnerKind;
+	}
+
+	public void setContainerHandle(String containerHandle) {
+		fContainerHandle= containerHandle;
+	}
+
+
+	@Override
+	public String getAttribute(String attributeName, String defaultValue) throws CoreException {
+		if (IJavaLaunchConfigurationConstants.ATTR_PROJECT_NAME.equals(attributeName)) {
+			return fProjectName;
+		} else if (JUnitLaunchConfigurationConstants.ATTR_TEST_RUNNER_KIND.equals(attributeName)) {
+			return fTestRunnerKind;
+		} else if (IJavaLaunchConfigurationConstants.ATTR_MAIN_TYPE_NAME.equals(attributeName)) {
+			return fMainTypeName;
+		} else if (JUnitLaunchConfigurationConstants.ATTR_TEST_CONTAINER.equals(attributeName)) {
+			return fContainerHandle;
+		}
+		return defaultValue;
+	}
+
+	@Override
+	public List<String> getAttribute(String attributeName, List<String> defaultValue) throws CoreException {
+		return defaultValue;
+	}
+
+	@Override
+	public boolean getAttribute(String attributeName, boolean defaultValue) throws CoreException {
+		return defaultValue;
+	}
+
+	@Override
+	public <T> T getAdapter(Class<T> adapter) {
+		return null;
+	}
+
+	@Override
+	public boolean contentsEqual(ILaunchConfiguration configuration) {
+		return false;
+	}
+
+	@Override
+	public ILaunchConfigurationWorkingCopy copy(String name) throws CoreException {
+		return null;
+	}
+
+	@Override
+	public void delete() throws CoreException {
+	}
+
+	@Override
+	public void delete(int flag) throws CoreException {
+	}
+
+	@Override
+	public boolean exists() {
+		return false;
+	}
+
+	@Override
+	public int getAttribute(String attributeName, int defaultValue) throws CoreException {
+		return 0;
+	}
+
+	@Override
+	public Set<String> getAttribute(String attributeName, Set<String> defaultValue) throws CoreException {
+		return null;
+	}
+
+	@Override
+	public Map<String, String> getAttribute(String attributeName, Map<String, String> defaultValue) throws CoreException {
+		return null;
+	}
+
+	@Override
+	public Map<String, Object> getAttributes() throws CoreException {
+		return null;
+	}
+
+	@Override
+	public String getCategory() throws CoreException {
+		return null;
+	}
+
+	@Override
+	public IFile getFile() {
+		return null;
+	}
+
+	@Override
+	public IPath getLocation() {
```

Comment:
please fix or suppress this warning:
```
Deprecation
The method MockLaunchConfig.getLocation() overrides a deprecated method from ILaunchConfiguration
```

Reply:
everything else looks good.

Reply:
I've changed the code to suppress the warning.